tests: TestXLStorageDeleteFile sporadic failure

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
Unit tests always succeed

## Current Behavior
```
go test -run=TestXLStorageDeleteFile -count=1000 .
--- FAIL: TestXLStorageDeleteFile (0.00s)
    xl-storage_test.go:1073: Expected: "Disk not found", got "unformatted disk found"
--- FAIL: TestXLStorageDeleteFile (0.00s)
    xl-storage_test.go:1073: Expected: "Disk not found", got "unformatted disk found"
FAIL
FAIL    github.com/minio/minio/cmd      3.902s
FAIL
```

## Possible Solution
Not yet 

## Steps to Reproduce (for bugs)
go test -run=TestXLStorageDeleteFile -count=1000 .

## Context
Fix unit tests

--- a/cmd/prepare-storage.go
+++ b/cmd/prepare-storage.go
@@ -70,7 +70,7 @@ var printEndpointError = func() func(Endpoint, error, bool) {
 }()
 
 // Cleans up tmp directory of the local disk.
-func formatErasureCleanupTmp(diskPath string) {
+func bgFormatErasureCleanupTmp(diskPath string) {
 	// Need to move temporary objects left behind from previous run of minio
 	// server to a unique directory under `minioMetaTmpBucket-old` to clean
 	// up `minioMetaTmpBucket` for the current run.
@@ -98,9 +98,8 @@ func formatErasureCleanupTmp(diskPath string) {
 	}
 
 	go removeAll(tmpOld)
-
 	// Renames and schedules for purging all bucket metacache.
-	renameAllBucketMetacache(diskPath)
+	go renameAllBucketMetacache(diskPath)
 }
 
 // Following error message is added to fix a regression in release

--- a/cmd/xl-storage.go
+++ b/cmd/xl-storage.go
@@ -244,7 +244,7 @@ func newXLStorage(ep Endpoint) (s *xlStorage, err error) {
 		diskIndex:  -1,
 	}
 
-	go formatErasureCleanupTmp(s.diskPath) // cleanup any old data.
+	bgFormatErasureCleanupTmp(s.diskPath) // cleanup any old data.
 
 	formatData, formatFi, err := formatErasureMigrate(s.diskPath)
 	if err != nil && !errors.Is(err, os.ErrNotExist) {

--- a/cmd/xl-storage_test.go
+++ b/cmd/xl-storage_test.go
@@ -959,13 +959,6 @@ func TestXLStorageDeleteFile(t *testing.T) {
 	}
 	defer os.RemoveAll(path)
 
-	// create xlStorage test setup
-	xlStorageDeletedStorage, diskPath, err := newXLStorageTestSetup()
-	if err != nil {
-		t.Fatalf("Unable to create xlStorage test setup, %s", err)
-	}
-	// removing the disk, used to recreate disk not found error.
-	os.RemoveAll(diskPath)
 	// Setup test environment.
 	if err = xlStorage.MakeVol(context.Background(), "success-vol"); err != nil {
 		t.Fatalf("Unable to create volume, %s", err)
@@ -1064,6 +1057,17 @@ func TestXLStorageDeleteFile(t *testing.T) {
 		if err = xlStorageNew.Delete(context.Background(), "mybucket", "myobject", false); err != errVolumeAccessDenied {
 			t.Errorf("expected: %s, got: %s", errVolumeAccessDenied, err)
 		}
+	}
+
+	// create xlStorage test setup
+	xlStorageDeletedStorage, diskPath, err := newXLStorageTestSetup()
+	if err != nil {
+		t.Fatalf("Unable to create xlStorage test setup, %s", err)
+	}
+	// removing the disk, used to recreate disk not found error.
+	err = os.RemoveAll(diskPath)
+	if err != nil {
+		t.Fatalf("Unable to remoe xlStorage diskpath, %s", err)
 	}
 
 	// TestXLStorage for delete on an removed disk.

